protocol/xmrmaker/net.go

```diff
@@ -73,6 +70,9 @@ func (b *Instance) initiate(
 
 // HandleInitiateMessage is called when we receive a network message from a peer that they wish to initiate a swap.
 func (b *Instance) HandleInitiateMessage(msg *net.SendKeysMessage) (net.SwapState, net.Message, error) {
+	b.swapMu.Lock()
+	defer b.swapMu.Unlock()
```

Comment:
So these were the missing lines that were needed to prevent the race condition when 2 parties try to take the same offer simultaneously?

Reply:
maybe, I don't think it fully fixes it, I think the issue is more how the test is written or another issue - when I looked at the logs, the issue was that one of the takers was hanging.